core/src/main/java/com/google/instrumentation/trace/Span.java

```diff
@@ -46,17 +48,25 @@
     RECORD_EVENTS;
   }
 
+  private static final EnumSet<Options> DEFAULT_OPTIONS = EnumSet.noneOf(Options.class);
+
   /**
    * Creates a new {@code Span}.
    *
    * @param context The context associated with this {@code Span}.
    * @param options The options associated with this {@code Span}. If {@code null} then default
    *     options will be set.
    * @throws NullPointerException if context is {@code null}.
+   * @throws IllegalArgumentException is the {@code SpanContext} is sampled but no RECORD_EVENTS
+   *     options.
    */
   protected Span(SpanContext context, @Nullable EnumSet<Options> options) {
     this.context = checkNotNull(context, "context");
-    this.options = options == null ? EnumSet.noneOf(Options.class) : EnumSet.copyOf(options);
+    this.options =
+        Collections.unmodifiableSet(options == null ? DEFAULT_OPTIONS : EnumSet.copyOf(options));
```

Comment:
Suggestion: I think it's simpler to use Collections.unmodifiableSet to create DEFAULT_OPTIONS and then have:
```java
 this.options =
         options == null ? DEFAULT_OPTIONS : Collections.unmodifiableSet(EnumSet.copyOf(options));
```